### DISCUSSION TITLE
Why the tags filter is not work when I search log level In u

### DISCUSSION THREAD

**Discussion by zhangzx1996** [apache/skywalking#12426] [Q&A] [answered]
![image](https://github.com/apache/skywalking/assets/46373828/1f6218de-4ed0-4c5e-a9e0-6cb7c64ae70e)

![image](https://github.com/apache/skywalking/assets/46373828/79ba4493-188a-4173-8f13-40852b59dbb2)

If I put level=INFO filter, no log can be searched!

**Comment by wu-sheng** [comment, marked as answer]
Refer to this in the `application.yml`

```
    # Define the set of log tag keys, which should be searchable through the GraphQL.
    # The max length of key=value should be less than 256 or will be dropped.
    searchableLogsTags: ${SW_SEARCHABLE_LOGS_TAG_KEYS:level,http.status_code}
```

So `logger` is omitted. But `level` is.

**Reply by zhangzx1996** [reply]
Very thanks. I missed this config.

**Comment by wu-sheng** [comment]
Refer to this in the `application.yml`

```
    # Define the set of log tag keys, which should be searchable through the GraphQL.
    # The max length of key=value should be less than 256 or will be dropped.
    searchableLogsTags: ${SW_SEARCHABLE_LOGS_TAG_KEYS:level,http.status_code}
```

So `logger` is omitted. But `level` is.

**Comment by wankai123** [comment]
I tried on the DEMO, and it works. 
<img width="1036" alt="image" src="https://github.com/apache/skywalking/assets/16773043/dfcdae38-0078-4fad-9bf6-9db1d3d5d97b">

**Comment by zhangzx1996** [comment]
I found some question about this again. I can search logs which the level is error, but Info can't.

**Reply by zhangzx1996** [reply]
I have set the config:
SW_SEARCHABLE_LOGS_TAG_KEYS: level

**Reply by zhangzx1996** [reply]
<img width="1280" alt="image" src="https://github.com/user-attachments/assets/43bd07dc-6665-41a8-b4dd-4064d5ac533b">

**Reply by zhangzx1996** [reply]
<img width="1356" alt="image" src="https://github.com/user-attachments/assets/66599fa7-735e-4a03-b8cb-e09b662a2663">

**Reply by zhangzx1996** [reply]
@wu-sheng  Can you help me? Very thanks.

**Reply by wu-sheng** [reply]
Do you ever push the INFO logs from your application? THere is nothing special. If there is no record, then this.

**Reply by zhangzx1996** [reply]
> Do you ever push the INFO logs from your application? THere is nothing special. If there is no record, then this.

<img width="1071" alt="image" src="https://github.com/user-attachments/assets/8861679a-87e7-46d6-a7d8-7c7b22d71693">
Yes, there are many Info logs. but I can't get them when I set Info level fillter.

**Reply by wankai123** [reply]
What storage are you using？ Elasticsearch?

**Reply by wankai123** [reply]
And make sure the logs in your query time duration

**Reply by zhangzx1996** [reply]
> What storage are you using？ Elasticsearch?

Yes, ES

**Reply by zhangzx1996** [reply]
> And make sure the logs in your query time duration

My query time duration is correct. I can get Info logs without Info level fillter

**Reply by wankai123** [reply]
could you put your Browser logs graphQL query network info here? include request and response.

**Reply by zhangzx1996** [reply]
> could you put your Browser logs graphQL query network info here? include request and response.

How to get the graphQL?

**Reply by wankai123** [reply]
The http query info in the Browser which requests the graphql

**Reply by zhangzx1996** [reply]
> The http query info in the Browser which requests the graphql

{
  "query": "query queryLogs($condition: LogQueryCondition) {\n    queryLogs(condition: $condition) {\n        logs {\n          serviceName\n          serviceId\n          serviceInstanceName\n          serviceInstanceId\n          endpointName\n          endpointId\n          traceId\n          timestamp\n          contentType\n          content\n          tags {\n            key\n            value\n          }\n        }\n    }}",
  "variables": {
    "condition": {
      "queryDuration": {
        "start": "2024-07-23 0530",
        "end": "2024-07-23 0600",
        "step": "MINUTE"
      },
      "paging": {
        "pageNum": 1,
        "pageSize": 15
      },
      "serviceId": "c2VsZjo6cGF5bWVudC1jb3Jl.1",
      "keywordsOfContent": [],
      "excludingKeywordsOfContent": [],
      "tags": [
        {
          "key": "level ",
          "value": " INFO"
        }
      ]
    }
  }
}

**Reply by zhangzx1996** [reply]
> key

I can't get INFO logs , only can filter ERROR logs. but there are many INFO logs

**Reply by zhangzx1996** [reply]
> > The http query info in the Browser which requests the graphql
> 
> { "query": "query queryLogs($condition: LogQueryCondition) {\n queryLogs(condition: $condition) {\n logs {\n serviceName\n serviceId\n serviceInstanceName\n serviceInstanceId\n endpointName\n endpointId\n traceId\n timestamp\n contentType\n content\n tags {\n key\n value\n }\n }\n }}", "variables": { "condition": { "queryDuration": { "start": "2024-07-23 0530", "end": "2024-07-23 0600", "step": "MINUTE" }, "paging": { "pageNum": 1, "pageSize": 15 }, "serviceId": "c2VsZjo6cGF5bWVudC1jb3Jl.1", "keywordsOfContent": [], "excludingKeywordsOfContent": [], "tags": [ { "key": "level ", "value": " INFO" } ] } } }

This is Response
`{"data":{"queryLogs":{"logs":[]}}}`

**Reply by wu-sheng** [reply]
I don't think it is. Because there is no special process for the value. 
If it is there, then you will have that.

**Reply by zhangzx1996** [reply]
> I don't think it is. Because there is no special process for the value.

But I can get many INFO logs without level filter. and I can get ERROR level logs with ERROR filter, others is not!

**Reply by wu-sheng** [reply]
Then your original logs are reported by error or some other level tags.
You need to check your raw loga and data in the database. 

You asked this for days, why don't you take an action to try to read data?

**Reply by zhangzx1996** [reply]
> Then your original logs are reported by error or some other level tags. You need to check your raw loga and data in the database.
> 
> You asked this for days, why don't you take an action to try to read data?

The tags in ES is encoded, and My data is correct!

**Reply by wu-sheng** [reply]
If you don't want to take actions, be yourself.
We can't debug for you.
